### PR TITLE
Custom transaction fails to validate if `FEE` member is not defined - Closes #3941

### LINKED ISSUE #3941: Custom transaction fails to validate if `FEE` member is not defined
### Expected behavior
Expect custom transaction to not crash during validation if `FEE` member is not defined.

### Actual behavior
The PR https://github.com/LiskHQ/lisk-sdk/pull/3893 introduced a new static member called `FEE` to `BaseTransaction`. This property should be implemented by custom transactions to define its own fee but it crashes during validation if it's not defined.

### Steps to reproduce
Create a custom transaction extending `BaseTransaction` like:
```
class CustomTransaction extends BaseTransaction {
	static get TYPE () {
		return 10;
	}

	validateAsset() {
		return[];
	}
}
```
Then:
```
const tx = new CustomTransaction({
  recipientId: '10881167371402274308L',
});

tx.sign('passphrase');
tx.validate();
```

`validate()` will throw `eq() not a number: undefined`

### Which version(s) does this affect? (Environment, OS, etc...)
2.1

### PR DESCRIPTION
**Pull request by lsilvs**
### What was the problem?
Custom transaction crashing during validation if FEE member is not defined.

### How did I fix it?
By setting a default value for `FEE`

### How to test it?
Create a custom transaction extending `BaseTransaction` like:
```
class CustomTransaction extends BaseTransaction {
	static get TYPE () {
		return 10;
	}

	validateAsset() {
		return[];
	}
}
```
Then:
```
const tx = new CustomTransaction({
  recipientId: '10881167371402274308L',
});

tx.sign('passphrase');
tx.validate();
```

It should return successful transaction response

### Review checklist

* The PR resolves #3941
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: elements/lisk-transactions/test/base_transaction.ts [hunk 1]
```diff
@@ -501,6 +504,15 @@ describe('Base transaction class', () => {
 			expect(status).to.eql(Status.OK);
 		});
 
+		it('should return a successful transaction response with a valid transaction with basic impl', async () => {
+			transactionWithBasicImpl.sign('passphrase');
```
Comment (shuse2) [inline elements/lisk-transactions/test/base_transaction.ts]: shouldn't we check if we add the invalid fee, it becomes `0`?

File: elements/lisk-transactions/src/base_transaction.ts [hunk 2]
```diff
@@ -115,7 +115,7 @@ export abstract class BaseTransaction {
 	public receivedAt?: Date;
 
 	public static TYPE: number;
-	public static FEE: string;
+	public static FEE = '0';
```
[the remark this answers is not in the document]
Comment (lsilvs) [reply elements/lisk-transactions/src/base_transaction.ts]: We don't have default value for `TYPE`

File: elements/lisk-transactions/test/base_transaction.ts (hunk 1 above)
Comment (lsilvs) [reply elements/lisk-transactions/test/base_transaction.ts]: I've added a new test case scenario ;)

File: elements/lisk-transactions/src/base_transaction.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lsilvs) [reply elements/lisk-transactions/src/base_transaction.ts]: ah! You mean variable type, not transaction type? haha

If so, tslint said I shouldn't use
```
no-inferrable-types -> Type string trivially inferred from a string literal, remove type annotation
```

File: elements/lisk-transactions/src/base_transaction.ts [hunk 3]
```diff
@@ -143,13 +143,22 @@ export abstract class BaseTransaction {
...
+
 		this.fee = new BigNum(
-			isValidNumber(tx.fee) ? (tx.fee as string | number) : '0',
+			isValidNumber(tx.fee)
+				? (tx.fee as string | number)
+				: (this.constructor as typeof BaseTransaction).FEE,
 		);
 
+		this.type =
+			typeof tx.type === 'number'
```
Comment (pablitovicente) [inline elements/lisk-transactions/src/base_transaction.ts]: Maybe not to be fixed in this PR but should we discuss a way to let the user know we're defaulting to something or maybe create an issue to document this behaviour?

**Pull request merged**